src/main/kotlin/com/atlassian/performance/tools/report/api/judge/Verdict.kt

```diff
@@ -7,17 +7,15 @@ import com.atlassian.performance.tools.report.api.junit.SuccessfulJUnitReport
 import org.apache.logging.log4j.LogManager
 import java.nio.file.Path
 
-class Verdict internal constructor(
-    private val reports: List<JUnitReport>,
-    val failedActions: List<ActionType<*>>
+class Verdict(
+    val reports: List<JUnitReport>
 ) {
 
-    constructor(
-        reports: List<JUnitReport>
-    ) : this(
-        reports = reports,
-        failedActions = emptyList()
-    )
+    @Deprecated("Use reports instead.")
+    val failedActions: List<ActionType<*>>  = reports
+        .filterIsInstance<ActionReport>()
```

Comment:
Again, implementation sniffing is a huge problem. It forbids delegating implementations and therefore aspect-oriented programming. AOP allows users to add new features, but also work around bugs.

Reply:
But I need to preserve previous behavior. Also, it will be removed anyway.

Reply:
To unlock delegating implementations we need to promote interfaces like this change does.

Reply:
> it will be removed anyway

I accept this. If only the old API is doing that, then it's fine, because users can switch to the new API and reap the AOP benefits.
But, will the new API avoid implementation sniffing? I don't see how a `Verdict` consumer can reliably access the  `ActionReport` data.

Reply:
@drspartez you're right, we'd need to promote the interfaces. But `Verdict` still returns `JUnitReport`, it's not promoted. And promoting that will add new requirements to all judges, even those, who are action-agnostic.
So it seems the current `Verdict` should work perfectly. I don't understand the need for this PR.

Reply:
We need to know if there was any "critical" error before we assert. We also need to be able to dump reports without asserting. Dump operation is combined with the assertion which is wrong IMO. Currently, we cannot do either of that with `Verdict`.

Reply:
You're right - the new API will not let consumers reliably access `ActionReport` data without implementation sniffing, but I don't understand why it is "a huge problem".

Reply:
Thank you, now it's much clearer.
Shouldn't we then simply add a `dump` method to `Verdict` and API for `critical`?
PS. What does `critical = true` mean?

Reply:
> why it is "a huge problem"

I made [this mistake before](https://bitbucket.org/atlassian/aws-infrastructure/pull-requests/58/issue-jperf-357-reuse-vpcs/activity#comment-96032308).

Reply:
Oh, I forgot one thing: we want to be able to merge different `Verdict`s

Reply:
Would it be possible to do via the existing `plus` method?

Reply:
No, we need the flexibility, we need access to the list of `ActionReport`s

Reply:
I don't see the contradiction. Why would `plus` be unable to access that list?